packages/react-components/react-color-picker-preview/library/docs/Spec.md

```diff
@@ -2,62 +2,458 @@
+| min          | `number`                 | 0            | The min value of the Slider.           |
+| onChange     | `function`               | undefined    | Callback called when color is selected |
+| orient       | `horizontal`, `vertical` | `horizontal` | Orientation of a slider                |
+| overlayColor | `string`                 | undefined    | Overlay color                          |
+| shape        | `square`, `rounded`      | `square`     | Sets shape                             |
+| step         | `number`                 | 1            | Step for the slider                    |
+| value        | `string`                 |              | value of the slider                    |
+
+| Slots  | Values  | Default | Description                                                      |
+| ------ | ------- | ------- | ---------------------------------------------------------------- |
+| root   | `div`   | `div`   | The root of the AlphaSlider element                              |
+| thumb  | `div`   | `div`   | The draggable icon used to select a given value from the Slider. |
+| slider | `input` | `input` | Input for slider                                                 |
+| rail   | `div`   | `div`   | It is used to visibly display the min and max selectable values. |
 
 ## Structure
 
-- _**Public**_
-- _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
+### Components
+
+| Component   | Purpose                                      |
+| ----------- | -------------------------------------------- |
+| ColorPicker | Renders ColorPicker                          |
+| ColorArea   | Renders two-dimensional gradient background. |
+| ColorSlider | Renders individual color channel             |
+| AlphaSlider | Renders slider with alpha channel            |
+
+#### ColorPicker component
+
+![ColorPicker Anatomy](./assets/color-picker-anatomy.jpg)
+
+#### DOM
+
+```HTML
+<div role="group" class="fui-ColorPicker" arial-label="ColorPicker">
+  <!-- Content rendered here -->
+</div>
+```
+
+#### ColorArea component
+
+![ColorArea Anatomy](./assets/color-area-anatomy.jpg)
+
+#### DOM
+
+```HTML
+<div role="group" class="fui-ColorArea" aria-label="Saturation and brightness" aria-valuetext="Saturation 54 brightness 35">
+  <input id="sliderX" type="range" class="fui-ColorArea__inputX" value="96" role="slider">
+  <input id="sliderY" type="range" class="fui-ColorArea__inputY" value="0" role="slider">
+  <div class="fui-ColorArea__thumb" role="presentation"></div>
+</div>
+```
+
+#### ColorSlider component
+
+##### Horizontal
+
+![ColorSlider horizontal Anatomy](./assets/color-slider-horizontal.jpg)
+
+##### Vertical
+
+![ColorSlider vertical Anatomy](./assets/color-slider-vertical.jpg)
+
+#### DOM
+
+```HTML
+<div class="fui-ColorSlider" role="group">
+  <input id="slider" type="range" role="slider" value="100" aria-valuetext="100, red" max="360">
```

Comment:
the native input already has the right role, so you can actually remove this 😊 (also on the other inputs as well, I missed that in the other comment)
```suggestion
  <input id="slider" type="range" value="100" aria-valuetext="100, red" max="360">
```

Reply:
Just noticed another thing -- this input would need a label as well. I think you could actually drop the `aria-valuetext` here too, and make the label `aria-label="red"`